### ISSUE TITLE
Poco don't compile on Debian strecht testing 

### ISSUE THREAD

**Issue by yoosamui** [openframeworks/openFrameworks#5397]
of_v0.9.8_linux64_release
after run install_dependencies.sh (linux/debian)  i get errors:

1.
src/DigestEngine.cpp: In member function 'int Poco::Crypto::DigestEngine::nid() const':
src/DigestEngine.cpp:42:24: error: invalid use of incomplete type 'EVP_MD_CTX {aka struct evp_md_ctx_st}'
  return EVP_MD_nid(_ctx->digest);

2.
src/DigestEngine.cpp: In member function 'virtual void Poco::Crypto::DigestEngine::reset()':
src/DigestEngine.cpp:53:25: error: 'EVP_MD_CTX_cleanup' was not declared in this scope
  EVP_MD_CTX_cleanup(_ctx);

3.
src/CipherImpl.cpp:80:21: error: field '_ctx' has incomplete type 'EVP_CIPHER_CTX {aka evp_cipher_ctx_st}'
   EVP_CIPHER_CTX    _ctx;

4.
src/RSAKeyImpl.cpp: In member function 'Poco::Crypto::RSAKeyImpl::ByteVec Poco::Crypto::RSAKeyImpl::modulus() const':
src/RSAKeyImpl.cpp:210:31: error: invalid use of incomplete type 'RSA {aka struct rsa_st}'
  return convertToByteVec(_pRSA->n);
                               ^~
5.
src/RSAKeyImpl.cpp: In member function 'Poco::Crypto::RSAKeyImpl::ByteVec Poco::Crypto::RSAKeyImpl::modulus() const':
src/RSAKeyImpl.cpp:210:31: error: invalid use of incomplete type 'RSA {aka struct rsa_st}'
  return convertToByteVec(_pRSA->n);
                               ^~
6.
src/X509Certificate.cpp: In constructor 'Poco::Crypto::X509Certificate::X509Certificate(X509*, bool)':
src/X509Certificate.cpp:62:9: error: invalid use of incomplete type 'X509 {aka struct x509_st}'

 ^ Received error ^
i run the ubuntu install_dependencies.sh script but also finish with errors...
Any ideas ! Thanks.

**Comment by arturoc** [comment]
yes, this is a known problem, the current version of poco is not compatible with debian testing anymore. this won't be fixed in 0.9.x since 0.10 is almost ready but the nightly builds from today on will work without problem. be sure to use the linux64 gcc6 version

**Comment by yoosamui** [comment]
**# I can't compile poco with debian:**

Description:	Debian GNU/Linux 9.0 (stretch)
Release:	9.0
Codename:	stretch
Kernel:  Linux osx 4.9.0-1-amd64 #1 SMP Debian 4.9.6-3 

i used OP : of_v20170215_linux64gcc6_nightly.tar.gz
here is the output after run  install_dependencies.sh:

```
It seems you are running gcc 5 or later, due to incomatible ABI with previous versions
we need to recompile poco. This will take a while
./install_dependencies.sh: 79: read: Illegal option -n
./install_dependencies.sh: 88: ./install_dependencies.sh: Bad substitution
./install_dependencies.sh: 89: cd: can't cd to /home/yoo/Develop/of_v20170215_linux64gcc6_release/scripts/linux/debian/../../apothecary/apothecary
./install_dependencies.sh: 90: ./install_dependencies.sh: ./apothecary: not found


I have all compilers versions:
yoo@osx:~/Develop/of_v20170215_linux64gcc6_release/scripts/linux/debian$ dpkg --list | grep compiler
ii  g++                                   4:6.3.0-1                            
ii  g++-6                                 6.3.0-6                              
ii  gcc                                   4:6.3.0-1                            
ii  gcc-6                                 6.3.0-6                              
ii  libllvm3.9:amd64                      1:3.9.1-4                    

I have no idea what i can do.
Thank you for any help.
